subprojects/dependency-management/src/main/java/org/gradle/api/internal/artifacts/ivyservice/ivyresolve/DefaultVersionedComponentChooser.java

```diff
@@ -68,51 +68,55 @@ private boolean isGeneratedModuleDescriptor(ComponentResolveMetadata componentRe
         return componentResolveMetadata.isGenerated();
     }
 
-    public void selectNewestMatchingComponent(Collection<? extends ModuleComponentResolveState> versions, BuildableComponentSelectionResult result, ModuleVersionSelector requested) {
+    public void selectNewestMatchingComponent(Collection<? extends ModuleComponentResolveState> versions, ComponentSelectionContext result, ModuleVersionSelector requested) {
```

Comment:
If we go with comments, then I'd much prefer a description of the algorithm as a method-level comment instead of little snippets sprinkled over the method.